src/app/view-reviews/components/ReviewCardFilterModal.tsx

```diff
@@ -0,0 +1,302 @@
+    setParent(cat);
+    setSubcatMode("auto");
+    setSpecificSubs([]);
+  };
+
+  const childList = parent
+    ? allCategories.filter((c) => c.parent_id === parent.id)
+    : [];
+
+  const handleSelectSub = (sub: Category) => {
+    setSpecificSubs([sub]);
+    setSubcatMode("specific");
+  };
+
+  const handleApply = () => {
+    let categoryFilter: string[] | null;
+
+    if (!parent) {
+      // Case 2: no parent selected
+      categoryFilter = null;
+    } else {
+      const children = childList;
+      if (children.length === 0) {
+        // Case 1: parent with no children
+        categoryFilter = [parent.name];
+      } else {
+        switch (subcatMode) {
+          case "auto":
+            // Case 3: parent + automatic subcategories
+            categoryFilter = [parent.name, ...children.map((c) => c.name)];
+            break;
+          case "all":
+            // Case 5: parent + all subcategories
+            categoryFilter = [parent.name, ...children.map((c) => c.name)];
+            break;
+          case "none":
+            // Case 6: parent + no subcategories
+            categoryFilter = [parent.name];
+            break;
+          case "specific":
+            // Case 4: parent + specific subcategories
+            categoryFilter = specificSubs.map((c) => c.name);
+            break;
+        }
+      }
+    }
+
+    onApply({
+      rating: ratingFilter,
+      category: categoryFilter,
+    });
+    onDismiss();
+  };
+
+  const handleReset = () => {
+    const resetFilters = {
+      rating: { lower: 0, upper: 5 },
+      category: null,
+    };
+
+    setRating(resetFilters.rating);
+    setParent(null);
+    setSubcatMode("auto");
+    setSpecificSubs([]);
+    onApply(resetFilters);
+    onDismiss();
+  };
+
+  return (
+    <IonModal
+      isOpen={isOpen}
+      onDidDismiss={onDismiss}
+      ref={modal}
+      initialBreakpoint={0.80}
+      breakpoints={[0, 0.80]}
+      className="modal-filter"
+    >
+      <IonHeader className="ion-no-border ion-padding-top">
+        <IonToolbar>
+          <IonTitle className="text-2xl font-bold text-start ion-padding">
+            {t("review-page.filter")}
+          </IonTitle>
+        </IonToolbar>
+      </IonHeader>
+      <IonContent className="ion-padding ">
+        <div className="pb-5 w-full">
+          <div className="flex gap-x-4 items-center overflow-x-auto pb-3 pt-2 ps-2">
+            <div
+              className="flex flex-col items-center gap-2 justify-center w-23"
+              onClick={() => handleSelectParent(null)}
+            >
+              <div
+                className={`size-23 rounded-2xl flex items-center bg-[var(--ion-color-secondary)] justify-center ${
+                  !parent ? "selected" : ""
+                }`}
+              >
+                <Boxes size={40}></Boxes>
+              </div>
+              <IonLabel className="truncate max-w-full text-xs">{t('review-page.all')}</IonLabel>
```

Comment:
```suggestion
              <IonLabel className="truncate max-w-full text-xs">{t('common.all')}</IonLabel>
```

Esa traduccion tiene que estar en common, porque es mas generica